scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -159,6 +159,139 @@ trait JavaModule
    */
   def runIvyDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
 
+  /**
+   * Any BOM dependencies you want to add to this Module, in the format
+   * ivy"org:name:version"
+   */
+  def bomDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
+
+  def allBomDeps: Task[Agg[BomDependency]] = Task.Anon {
+    val modVerOrMalformed =
+      bomDeps().map(bindDependency()).map { bomDep =>
+        val fromModVer = coursier.core.Dependency(bomDep.dep.module, bomDep.dep.version)
+        if (fromModVer == bomDep.dep)
+          Right(bomDep.dep.asBomDependency)
+        else
+          Left(bomDep)
+      }
+
+    val malformed = modVerOrMalformed.collect {
+      case Left(malformedBomDep) =>
+        malformedBomDep
+    }
+    if (malformed.isEmpty)
+      modVerOrMalformed.collect {
+        case Right(bomDep) => bomDep
+      }
+    else
+      throw new Exception(
+        "Found BOM dependencies with invalid parameters:" + System.lineSeparator() +
+          malformed.map("- " + _.dep + System.lineSeparator()).mkString +
+          "Only organization, name, and version are accepted."
+      )
+  }
+
+  /**
+   * Dependency management data
+   *
+   * Versions and exclusions in dependency management override those of transitive dependencies,
+   * while they have no effect if the corresponding dependency isn't pulled during dependency
+   * resolution.
+   *
+   * For example, the following forces com.lihaoyi::os-lib to version 0.11.3, and
+   * excludes org.slf4j:slf4j-api from com.lihaoyi::cask that it forces to version 0.9.4
+   * {{{
+   *   def dependencyManagement = super.dependencyManagement() ++ Agg(
+   *     ivy"com.lihaoyi::os-lib:0.11.3",
+   *     ivy"com.lihaoyi::cask:0.9.4".exclude("org.slf4j", "slf4j-api")
+   *   )
+   * }}}
+   */
+  def dependencyManagement: T[Agg[Dep]] = Task { Agg.empty[Dep] }
```

Comment:
Should we name this `def depManagement`? We use `dep`/`Dep` throughout the rest of the codebase, so this would make the API a bit more consistent

Reply:
`depMgmt` could work too

Reply:
Went for `depManagement` for now